app/src/main/java/mozilla/lockbox/view/LockedFragment.kt

```diff
@@ -34,14 +42,27 @@ class LockedFragment : Fragment(), LockedView {
 
     override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
         super.onActivityResult(requestCode, resultCode, data)
+
+        log.info("onactivityresult, $resultCode")
         if (resultCode == RESULT_OK) {
             when (requestCode) {
-                Constant.RequestCode.unlock -> _unlockConfirmed.onNext(true)
+                Constant.RequestCode.lock -> _unlockConfirmed.onNext(true)
             }
         } else {
             _unlockConfirmed.onNext(false)
         }
     }
 
-    override val unlockConfirmed: Observable<Boolean> get() = _unlockConfirmed
+    override fun unlockFallback() {
+        val manager = context?.getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager
+        val intent = manager.createConfirmDeviceCredentialIntent(
+            getString(R.string.unlock_fallback_title),
+            getString(R.string.confirm_pattern)
+        )
+        try {
+            startActivityForResult(intent, Constant.RequestCode.lock)
+        } catch (exception: Exception) {
+            _unlockConfirmed.onNext(false)
+        }
+    }
```

Comment:
This should disappear, and be replaced with a `dispatcher.dispatch(RouteAction.UnlockFallbackDialog)` call in the presenter.